File: src/main/java/no/nav/syfo/scheduler/CacheEvictionScheduler.java
package no.nav.syfo.scheduler;

import org.springframework.cache.CacheManager;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import javax.inject.Inject;

@Component
public class CacheEvictionScheduler {

    private final int MINUTE_MS = 60 * 1000;
    private final int HOUR_MS = 60 * MINUTE_MS;

    private CacheManager cachemanager;

    @Inject
    public CacheEvictionScheduler(CacheManager cachemanager) {
        this.cachemanager = cachemanager;
    }

    @Scheduled(fixedRate = HOUR_MS)
    public void evictAllCachesAtInteval() {
        evictAllCaches();
    }

    private void evictAllCaches() {
        cachemanager.getCacheNames()
                .forEach(cacheName -> cachemanager.getCache(cacheName).clear());
    }
}

File: src/main/kotlin/no/nav/syfo/cache/CacheEvictionScheduler.kt
package no.nav.syfo.cache

import org.springframework.cache.CacheManager
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import java.util.function.Consumer
import javax.inject.Inject

@Component
class CacheEvictionScheduler @Inject constructor(
    private val cachemanager: CacheManager
) {
    @Scheduled(fixedRate = HOUR_MS.toLong())
    fun evictAllCachesAtInteval() {
        evictAllCaches()
    }

    private fun evictAllCaches() {
        cachemanager.cacheNames
            .forEach(Consumer { cacheName: String ->
                val cache = cachemanager.getCache(cacheName)
                cache?.clear()
            })
    }

    companion object {
        private const val MINUTE_MS = 60 * 1000
        private const val HOUR_MS = 60 * MINUTE_MS
    }
}
